ft/cachetable/cachetable-internal.h

```diff
@@ -498,12 +499,12 @@ class evictor {
 
     pair_list* m_pl;
     cachefile_list* m_cf_list;
-    int64_t m_size_current;            // the sum of the sizes of the pairs in the cachetable
+    std::atomic_int64_t m_size_current;            // the sum of the sizes of the pairs in the cachetable
```

Comment:
This makes 'class evictor' not a POD and triggers static assertion via 'ENSURE_POD(evictor);' in cachetable.cc around line 3611

Reply:
Actually, all of these atomic_type_t specializations are incorrect as they do not exist in all supported back versions of glib, must use std::atomic<type> instead.

Reply:
so for backward compatibility, use std::atomic<int> in place of std::atomic_int?

Reply:
yeah, these all need to be the older style specializations std::atomic<uint32_t>, etc... I easily changed those but the bigger issue is that by adding this to evictor (and possibly other class/structs) introduces non trivial layout and construction/destruction/copy semantics, which then causes failure of std::is_pod and triggers the static_assertion in ENSURE_POD. I do not recall offhand what the need for evictor and others to be POD was and what it would take to eliminate it.

Alternately, since the evictor is really just a singleton, a hack could be to just to move this to global scope, but that IMHO is going the wrong direction in the greater scheme to properly "C++-ize" the FT library.